frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -55,13 +86,50 @@ export function DynamicConfiguration () {
         formSubmitting: false as boolean,
     });
 
+    let formErrors = {} as any;
+    const parseResult = formSchema.safeParse(state.form);
+    if (!parseResult.success) {
+        formErrors = issuesToErrors(
+            parseResult.error.issues,
+            state.validateAll === true ? undefined : state.touched
+        );
+    }
+
     useEffect(() => {
         dispatch(listConfigurations({configScope: 'global', numVersions: 5}));
     }, [dispatch]);
 
+    useEffect(() => {
+        dispatch(listEMRApplications());
+    }, [dispatch]);
+
+    useEffect(() => {
+        const appList: any[] = [];
+        for (const application of applicationList) {
+            const app = {
+                value: application,
+                label: application
+            };
+            appList.push(app);
+        }
+        // Make sure we're always starting from an empty array to prevent duplicates
+        setSelectedApplicationOptions([]);
+        for (const configApp of applicationConfig.configuration.EMRConfig.applications) {
+            if (applicationList.includes(configApp.Name) && !selectedApplicationOptions.includes(configApp.Name)) {
+                const app = {
+                    value: configApp.Name,
+                    label: configApp.Name
+                };
+                setSelectedApplicationOptions((priorSelectedOptions) => [...priorSelectedOptions, app]);
+            }
+        }
+        setApplicationOptions(appList);
+        // eslint-disable-next-line react-hooks/exhaustive-deps 
+    }, [applicationList, applicationConfig]);
 
     const handleSubmit = async () => {
-        if (isValid) {
+        const parseResult = formSchema.safeParse(state.form);
+        if (parseResult.success) {
```

Comment:
Why change from `isValid` to checking manually? Isn't this doing the exact same thing as is happening in `useValidationReducer`?

Reply:
I think I was trying to make use of `formErrors`, I didn't see how we'd be doing that with `isValid` but if you know how we'd do that I'm happy to make the transition

Reply:
`useValidationReducer()` returns a [few things](https://github.com/awslabs/mlspace/blob/main/frontend/src/shared/validation/index.ts#L230-L237) including `isValid` and `errors` which I believe is [doing the same thing](https://github.com/awslabs/mlspace/blame/main/frontend/src/shared/validation/index.ts#L296-L303) as your code. So adding back `isValid` and additionally `errors` should give you exactly what.